src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -70,19 +81,133 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit v
     }
   }
 
-
-  def deleteCluster(googleProject: String, clusterName: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
+  override def deleteCluster(googleProject: String, clusterName: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
     Future {
       //currently, the bucketPath of the clusterRequest are not used - it will be used later as a place to store notebooks and results
-      val request = dataproc.projects().regions().clusters().delete(googleProject, dataprocConfig.dataprocDefaultZone, clusterName)
+      val request = dataproc.projects().regions().clusters().delete(googleProject, dataprocConfig.dataprocDefaultRegion, clusterName)
       try {
         executeGoogleRequest(request)
       } catch {
-        case e:GoogleJsonResponseException =>
-          if(e.getStatusCode!=404)
+        case e: GoogleJsonResponseException =>
+          if (e.getStatusCode != 404)
             throw CallToGoogleApiFailedException(googleProject, clusterName, e.getStatusCode, e.getDetails.getMessage)
       }
     }
   }
 
+  override def getClusterStatus(googleProject: GoogleProject, clusterName: String)(implicit executionContext: ExecutionContext): Future[ClusterStatus] = {
+    getCluster(googleProject, clusterName).map { cluster =>
+      ClusterStatus.withNameIgnoreCase(cluster.getStatus.getState)
+    }
+  }
+
+  override def getClusterMasterInstanceIp(googleProject: GoogleProject, clusterName: String)(implicit executionContext: ExecutionContext): Future[Option[String]] = {
+    val ipOpt = for {
+      cluster <- OptionT.liftF[Future, Cluster] { getCluster(googleProject, clusterName) }
```

Comment:
`OptionT` is handy when you want to deal with `Future[A]`, `Option[A]`, `Future[Option[A]]`, and `A` all in the same `flatMap`!

Reply:
That's weird and crazy!